### PR TITLE
[IMP] web: auto-select content on focus for m2o fields

### PR DESCRIPTION
**Pull request by caburj**
On first focus in an autocomplete field, the text content of the input element
is automatically highlighted.

TASK-ID: 3117420

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web/static/src/core/autocomplete/autocomplete.js [hunk 1]
```diff
@@ -228,6 +228,12 @@ export class AutoComplete extends Component {
         this.open(true);
     }
 
+    onInputFocus() {
+        if (this.inputRef.el) {
```
Comment (fdardenne) [inline addons/web/static/src/core/autocomplete/autocomplete.js:232]: For my knowledge: In which case is this condition needed ?

File: addons/web/static/src/core/autocomplete/autocomplete.xml [hunk 2]
```diff
@@ -15,6 +15,7 @@
                 t-on-change="onInputChange"
                 t-on-input="debouncedOnInput"
                 t-on-keydown="onInputKeydown"
+                t-on-focus="onInputFocus"
```
[the remark this answers is not in the document]
Comment (caburj) [reply addons/web/static/src/core/autocomplete/autocomplete.xml:18]: Oops. I should take that into account.

File: addons/web/static/src/core/autocomplete/autocomplete.js (hunk 1 above)
Comment (caburj) [reply addons/web/static/src/core/autocomplete/autocomplete.js]: True. I can remove the condition because the input ref el is always present.

**Review by aab-odoo — APPROVED** [review]
robodoo r+

**Pull request closed without merging**